packages/wrangler/src/triggers/deploy.ts

```diff
@@ -117,73 +132,78 @@ export default async function triggersDeploy(
 						return url;
 					})
 			);
-		} else {
-			deployments.push(Promise.resolve([scriptURL]));
 		}
-	} else {
-		if (available_on_subdomain) {
-			// Disable the workers.dev deployment
-			await fetchResult(`${workerUrl}/subdomain`, {
-				method: "POST",
-				body: JSON.stringify({ enabled: false }),
-				headers: {
-					"Content-Type": "application/json",
-				},
-			});
-		} else if (routes.length !== 0) {
-			// if you get to this point it's because
-			// you're trying to deploy a worker to a custom
-			// domain that's already bound to another worker.
-			// so this thing is about finding workers that have
-			// bindings to the routes you're trying to deploy to.
-			//
-			// the logic is kinda similar (read: duplicated) from publishRoutesFallback,
-			// except here we know we have a good API token or whatever so we don't need
-			// to bother with all the error handling tomfoolery.
-			const routesWithOtherBindings: Record<string, string[]> = {};
-			for (const route of routes) {
-				const zone = await getZoneForRoute({ route, accountId });
-				if (!zone) {
-					continue;
-				}
-
-				const routePattern = typeof route === "string" ? route : route.pattern;
-				const routesInZone = await fetchListResult<{
-					pattern: string;
-					script: string;
-				}>(`/zones/${zone.id}/workers/routes`);
+	}
+	if (!deployToWorkersDev && (!deploymentInSync || !previewsInSync)) {
+		// Disable the workers.dev deployment
+		await fetchResult(`${workerUrl}/subdomain`, {
+			method: "POST",
+			body: JSON.stringify({
+				enabled: false,
+				previews_enabled: config.workers_dev_previews,
+			}),
+			headers: {
+				"Content-Type": "application/json",
+			},
+		});
+	}
+	if (!deployToWorkersDev && deploymentInSync && routes.length !== 0) {
+		// TODO is this true? How does last subdomain status affect route confict??
```

Comment:
I kept this unchanged (aside from unrolling the conditional branches for easier context readability), but I don't think this branch makes sense anymore.